awswrangler/_utils.py

```diff
@@ -59,9 +60,13 @@ def boto3_from_primitives(primitives: Optional[Boto3PrimitivesType] = None) -> b
     return boto3.Session(**args)
 
 
-def botocore_config() -> botocore.config.Config:
+def default_botocore_config() -> botocore.config.Config:
     """Botocore configuration."""
-    return botocore.config.Config(retries={"max_attempts": 5}, connect_timeout=10, max_pool_connections=10)
+    retries_config: Dict[str, Union[str, int]] = {"max_attempts": int(os.getenv("AWS_MAX_ATTEMPTS", "5"))}
```

Comment:
this could probably be simplified to:

```py
retries_config = {
    "max_attempts": int(os.getenv("AWS_MAX_ATTEMPTS", "5")),
    "mode": os.getenv("AWS_RETRY_MODE", "legacy")
}
```

Reply:
I opted for the "complicated" version just to be sure that the retry mode will also be the default when `boto3/botocore` changes the default.

Reply:
Hm, the same could be said for `max_attempts` then - it's likely to change whenever `mode` is changed. Possibly also for `connect_timeout` and `max_pool_connections`.

Would be more consistent to either define all explicit defaults or fall back to `botocore` defaults entirely, unless there is a certain value that `awswrangler` benefits from.

Reply:
I'm also fine with setting it explicitly. Maybe it even makes more sense to directly use the `standard` mode even if it's not the default one. What do you think?

Reply:
I agree, seems like `mode=standard` makes more sense and is what I’ve used. The `botocore` default for `standard` is `max_attempts=3`, so that should be changed too.

Reply:
Changed it to `standard` mode and 3 attempts.